perception/traffic_light_occlusion_predictor/include/traffic_light_occlusion_predictor/nodelet.hpp

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TRAFFIC_LIGHT_OCCLUSION_PREDICTOR__NODELET_HPP_
+#define TRAFFIC_LIGHT_OCCLUSION_PREDICTOR__NODELET_HPP_
+
+#include <perception_utils/prime_synchronizer.hpp>
+#include <perception_utils/traffic_light_utils.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <traffic_light_occlusion_predictor/occlusion_predictor.hpp>
+
+#include <autoware_auto_mapping_msgs/msg/had_map_bin.hpp>
+#include <autoware_auto_perception_msgs/msg/traffic_light_occlusion_array.hpp>
+#include <autoware_auto_perception_msgs/msg/traffic_light_roi_array.hpp>
+#include <geometry_msgs/msg/pose_stamped.hpp>
+#include <sensor_msgs/msg/camera_info.hpp>
+#include <sensor_msgs/msg/point_cloud2.hpp>
+
+#include <image_geometry/pinhole_camera_model.h>
+#include <message_filters/subscriber.h>
+#include <message_filters/synchronizer.h>
+#include <message_filters/time_synchronizer.h>
+#include <tf2_ros/buffer.h>
+#include <tf2_ros/transform_listener.h>
+
+#include <map>
+#include <memory>
+#include <mutex>
+
+namespace traffic_light
+{
+class TrafficLightOcclusionPredictorNodelet : public rclcpp::Node
+{
+public:
+  explicit TrafficLightOcclusionPredictorNodelet(const rclcpp::NodeOptions & node_options);
+
+private:
+  struct Config
+  {
+    double azimuth_occlusion_resolution;
+    double elevation_occlusion_resolution;
+    double max_valid_pt_dist;
+    double max_image_cloud_delay;
+    double max_wait_t;
+  };
+
+private:
+  /**
+   * @brief receive the lanelet2 map
+   *
+   * @param input_msg
+   */
+  void mapCallback(const autoware_auto_mapping_msgs::msg::HADMapBin::ConstSharedPtr input_msg);
+  /**
+   * @brief subscribers
+   *
+   */
+  void syncCallback(
+    const autoware_auto_perception_msgs::msg::TrafficLightRoiArray::ConstSharedPtr in_roi_msg,
+    const sensor_msgs::msg::CameraInfo::ConstSharedPtr in_cam_info_msg,
+    const sensor_msgs::msg::PointCloud2::ConstSharedPtr in_cloud_msg);
+
+  rclcpp::Subscription<autoware_auto_mapping_msgs::msg::HADMapBin>::SharedPtr map_sub_;
+  /**
+   * @brief publishers
+   *
+   */
+  rclcpp::Publisher<autoware_auto_perception_msgs::msg::TrafficLightOcclusionArray>::SharedPtr
+    occlusion_pub_;
```

Comment:
[must] Could you change the output of occlusion predictor to `tier4_perception_msg/traffic_signals`?
![ArchitectureMain-traffic light ](https://github.com/autowarefoundation/autoware.universe/assets/37187849/32ea43f0-3060-47db-a3de-0d774b4325b4)

Reply:
@miursh 
The message type and output I/F are updated in commit [277411d](https://github.com/autowarefoundation/autoware.universe/pull/4050/commits/277411d1c6dc2856d5f9c27e5dadb3fea43b7427)